src/components/AmplitudeAnalytics/TraceEvent.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { PropsWithChildren } from 'react'
+import { SyntheticEvent } from 'react'
+
+import { sendAnalyticsEvent } from '.'
+import { EventName, PartialActionProps } from './constants'
+import { ITraceContext, Trace, TraceContext } from './Trace'
+
+type TraceEventProps = {
+  actionProps: PartialActionProps
+  eventName: EventName
+  eventProperties?: Record<string, unknown>
+} & ITraceContext
+
+/**
+ * Analytics instrumentation component that wraps event callbacks with logging logic.
+ *
+ * @example
+ *  <TraceEvent actionProps={{ onClick: { action: 'click' }}} elementType='button'>
+ *    <Button onClick={() => console.log('clicked')}>Click me</Button>
+ *  </TraceEvent>
+ */
+function _TraceEvent(props: PropsWithChildren<TraceEventProps>) {
+  const { eventName, eventProperties, actionProps, children, ...logEventProps } = props
```

Comment:
Rename logEventProps to traceProps so its usage is more clear

Reply:
done